src/extension.ts

```diff
@@ -230,11 +236,13 @@ export function hideStatusLine() {
 // find an output channel by name in the map and remove it
 function removeOutputChannel( channelName: string) {
 	let podOutputChannel;
-	if (outputChannelMap.has(channelName)) {
+	if (outputChannelMap !== undefined) {
 		podOutputChannel = outputChannelMap.get(channelName);
 		if (podOutputChannel !== undefined) {
-			podOutputChannel.dispose();
 			outputChannelMap.delete(channelName);
+			// concerned that this doesn't actually delete the resources associated with the output channel, 
+			// but dispose causes other problems
```

Comment:
what are the other problems with dispose?

Reply:
We were disposing the channel before (podOutputChannel.dispose(); on extension.ts line 236 or so) and for some reason there seems to be a thread somewhere still trying to update it in the background AFTER it's been disposed, which was what was creating the "channel has been closed" error I think. Dispose is what SHOULD be called it seems, but doesn't look like it's behaving quite as expected.

Reply:
it means that we are not disposing the pod output channel at all now, isn't it? If yes, sounds like a potential memory leak.

Reply:
We dispose of all output channels when the extension is deactivated. https://github.com/camel-tooling/vscode-camelk/blob/bce6fa9d91e9518cdb00c57372a6695da01d81ed/src/extension.ts#L473

But yes, other than that we are not disposing of them.

Reply:
the extensions not deactivated in a classic usage.
it means that we are trading errors in log against a memory leak. I prefer the errors in log.

Reply:
Ok. As the errors in the debug log don't affect the user, we'll just let it be then. Will close this PR